tracer/src/Datadog.Trace/OTelMetrics/OtlpExporter.cs

```diff
@@ -0,0 +1,468 @@
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Net.Sockets;
+using System.Threading;
+using System.Threading.Tasks;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Telemetry;
+
+#nullable enable
+
+namespace Datadog.Trace.OTelMetrics
+{
+    /// <summary>
+    /// OTLP Exporter implementation that exports metrics using the OpenTelemetry Protocol.
+    /// This is the concrete implementation of the MetricExporter base class that handles OTLP protocol specifics.
+    /// Supports both grpc and http/protobuf transports as specified in the RFC.
+    /// This is a Push Metric Exporter that sends metrics via the OpenTelemetry Protocol.
+    /// </summary>
+    internal class OtlpExporter : MetricExporter
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(OtlpExporter));
+        private readonly HttpClient _httpClient;
+        private readonly Telemetry.Metrics.MetricTags.Protocol _protocolTag;
+        private readonly OtlpMetricsSerializer _serializer;
+        private readonly Uri _endpoint;
+        private readonly IReadOnlyDictionary<string, string> _headers;
+        private readonly int _timeoutMs;
+        private readonly Configuration.OtlpProtocol _protocol;
+
+        public OtlpExporter(Configuration.TracerSettings settings)
+        {
+            _endpoint = settings.OtlpMetricsEndpoint;
+            _headers = settings.OtlpMetricsHeaders;
+            _timeoutMs = settings.OtlpMetricsTimeoutMs;
+            _protocol = settings.OtlpMetricsProtocol;
+
+            _protocolTag = _protocol switch
+            {
+                Configuration.OtlpProtocol.Grpc => Telemetry.Metrics.MetricTags.Protocol.Grpc,
+                Configuration.OtlpProtocol.HttpProtobuf => Telemetry.Metrics.MetricTags.Protocol.HttpProtobuf,
+                Configuration.OtlpProtocol.HttpJson => Telemetry.Metrics.MetricTags.Protocol.HttpJson,
+                _ => Telemetry.Metrics.MetricTags.Protocol.HttpProtobuf
+            };
+            _serializer = new OtlpMetricsSerializer(settings);
+
+            // Enable h2c (HTTP/2 without TLS) for gRPC only, and only for non-HTTPS endpoints
+            if (_protocol == Configuration.OtlpProtocol.Grpc && _endpoint.Scheme != Uri.UriSchemeHttps)
+            {
+                AppContext.SetSwitch("System.Net.Http.SocketsHttpHandler.Http2UnencryptedSupport", true);
+            }
+
+            _httpClient = CreateHttpClient(_endpoint);
+        }
+
+        /// <summary>
+        /// Exports a batch of metrics using OTLP protocol.
+        /// This method implements the MetricExporter base class Export operation.
+        /// </summary>
+        /// <param name="metrics">Batch of metrics to export</param>
+        /// <returns>ExportResult indicating success or failure</returns>
+        public override ExportResult Export(IReadOnlyList<MetricPoint> metrics)
```

Comment:
Can we delete this method? The reason I ask is the below `GetAwaiter().GetResult();` is risky, so if we don't need the code then let's remove it altogether. Otherwise, we will likely need to modify how you're calling sync over async code